moon/test_controller_round3.py

```diff
@@ -0,0 +1,89 @@
+import unittest
+from random import Random
+import math
+
+import numpy as np
+from shapely.geometry import LineString
+from shapely.geometry import Point
+
+from moon.controller_round3 import best_fit_circle
+
+def pol2cart(rho, phi):
+    x = rho * math.cos(phi)
+    y = rho * math.sin(phi)
+    return(x, y)
+
+def cart2pol(x, y):
+    rho = np.sqrt(x**2 + y**2)
+    phi = np.arctan2(y, x)
+    return(rho, phi)
+
+def plot_test_case(circle, lines):
+    circle_center_x, circle_center_y, circle_radius = circle
+    x_l, y_l = lines
+
+
+    fig, ax = plt.subplots()
+    ax.set_aspect(1)
+    ax.add_artist(plt.Circle((circle_center_x, circle_center_y), circle_radius))
+
+    for i in range(len(lines[0])):
+        plt.plot([0,x_l[i]],[0,y_l[i]])
+    plt.axis([-50, 50, -50, 50])
+    plt.show()
+
+has_matplotlib = False
```

Comment:
I would maybe rename it to `g_has_matplotlib` ... actually it should have different name like `g_display_debug_circles`??
(but feel free to ignore this comment)

Reply:
yes, the scan is ok, there are two intersections of the ray and the circle and currently we just take one arbitrarily and use that point, we do not look at the distance in order to pick the nearer one.